### PR TITLE
Enforce JDK 11 for development, but build Ingestion to target Java 8

### PR DESCRIPTION
**Pull request by ches**
**What this PR does / why we need it**:

This is a PoC for the idea in #517 that Ingestion could be easily built for Java 8 while other components continue to target 11.

```
$ mvn clean compile

$ javap -v ingestion/target/classes/feast/ingestion/ImportJob.class | grep major
  major version: 52

$ javap -v serving/target/classes/feast/serving/ServingApplication.class | grep major
  major version: 55
```

**Which issue(s) this PR fixes**:

Addresses #517, speculatively and not completely.

**Does this PR introduce a user-facing change?**:

```release-note
Feast Ingestion is again built for Java 8, for more certain Beam runner compatibility.
```

### CODE REVIEW

**Comment by ches** [comment]
Failure on `String#strip()` which is new in Java 11, guess that's a good sign 😇 Fixing…

**Comment by woop** [comment]
Thanks for this @ches. I've commented on the issue. Happy to lgtm/approve if you are done.

**Comment by ches** [comment]
> Happy to lgtm/approve if you are done.

Functionally this is working for me, with light experimentation. Assuming agreement on #517, I don't feel there's anything more to do on this branch except maybe add a note similar to the last part of https://github.com/gojek/feast/issues/517#issuecomment-596362212 to our development docs to clearly explain the policy and rationale.

It doesn't seem for me that IntelliJ automatically picked up the different `<release>` settings for maven-compiler-plugin to set [per-module language levels][1] automatically, I think that used to work for `<source>` and `<target>`. There may be some workflow or tool considerations like that to discover… interested to know if anyone finds any others.

https://www.jetbrains.com/help/idea/sources-tab.html#module_language_level

**Comment by woop** [comment]
@ches would you mind rebasing this one? I don't think the tests will pass otherwise.

**Comment by ches** [comment]
Done, hadn't responded to the bot yet because I wanted to come back to adding the development guide info anyway. Also done now.

File: docs/contributing/development-guide.md [hunk 1]
```diff
@@ -444,3 +444,35 @@ If you have made it to this point successfully you should have a functioning Fea
...
+
+See [Feast issue #517][\#517] for discussion.
+
+[the `javac --release` flag]: https://stackoverflow.com/questions/43102787/what-is-the-release-flag-in-the-java-9-compiler
+[BEAM-2530]: https://issues.apache.org/jira/browse/BEAM-2530
+[\#517]: https://github.com/gojek/feast/issues/517
+
+#### 5.2 IntelliJ Tips and Troubleshooting
+
+For IntelliJ users, this section collects notes and setup recommendations for working comfortably on the Feast project, especially to coexist as peacefully as possible with the Maven build.
```
Comment (ches) [inline docs/contributing/development-guide.md]: I'll move some other IntelliJ advice here that's duplicated (and out of sync) between the Contributing and Development guides, in another PR or GitBook edit.

I believe the thinking was that Contributing was getting unwieldy with technical setup / dev environment verification—I would agree, will take a look at moving things over with references left in Contributing. Probably this is also a legacy of duplicate `CONTRIBUTING.md` kept in the root awhile for GitHub.

Comment (woop) [reply docs/contributing/development-guide.md]: Yip, your assessment is correct. Lot's of little pebbles here, as you call them. We can chip away at it over time.

**Pull request merged**